include/vinecopulib/misc/tools_stats.hpp

```diff
@@ -100,12 +148,109 @@ simulate_uniform(const size_t& n,
                  bool qrng = false,
                  std::vector<int> seeds = std::vector<int>());
 
+Eigen::MatrixXd
+simulate_normal(const size_t& n,
+                const size_t& d,
+                std::vector<int> seeds = std::vector<int>());
+
 Eigen::VectorXd
 to_pseudo_obs_1d(Eigen::VectorXd x, const std::string& ties_method = "average");
 
 Eigen::MatrixXd
 to_pseudo_obs(Eigen::MatrixXd x, const std::string& ties_method = "average");
 
+class BoxCovering
+{
+public:
+  explicit BoxCovering(const Eigen::MatrixXd& u, uint16_t K = 40)
```

Comment:
Why uint16_t ? Shouldn't we use the same kind as in the grid for kernel bicops ?

Reply:
This saves many `static_cast<>`s when multiplying `K` or `K_` with `double` - in lines where we already cast the result of floor/ceil back to `size_t`. Simply multiplying `size_t * double` gave a STRICT_COMPILER fail on some of our CI builds. I can do the casts if you prefer.